sfge/src/main/java/com/salesforce/rules/AbstractRuleRunner.java

```diff
@@ -113,9 +113,7 @@ private Result runPathBasedRules(
         }
         Result res = ThreadableRuleExecutor.run(submissions);
         for (AbstractPathBasedRule rule : rules) {
-            if (rule instanceof PostProcessingRule) {
```

Comment:
I see a lot of this anti-pattern in the code. Maybe eventually we'll do a refactoring to clean all of these up. For now, I just touched this one since it was a low hanging fruit.

Reply:
Is there a name for that anti-pattern? Would love to understand better why it's undesirable.

Reply:
I don't want to get into a debate here, but I tend to live by the SOLID principles outlined in Robert Martin's books like "Clean Code",  "Clean Architecture", etc. Having conditional code on "types" is always something to question in an object oriented language.  I don't know if every anti-pattern has a name... but a quick google search will bring up the debate with https://stackoverflow.com/questions/30894002/is-it-good-practice-to-often-use-instanceof for example.

Reply:
I also just checked my "Refactorting: Improve the Design of Existing Code" book by Martin Fowler and on page 79 it talks briefly about this conditional smell... in which it proposes the solution "Replace Conditional with Polymorphism (on page 272)".